File: src/test/csharp/eu/sig/training/ch04/v3/SavingsAccountTest.cs
using NUnit.Framework;

namespace eu.sig.training.ch04.v3
{

    [TestFixture]
    public class SavingsAccountTest
    {

        SavingsAccount myAccount;
        CheckingAccount registeredCounterAccount;

        [SetUp]
        public void Init()
        {
            myAccount = Accounts.MakeAccount<SavingsAccount>("123456789");
            registeredCounterAccount = Accounts.MakeAccount<CheckingAccount>("497164833");
            myAccount.RegisteredCounterAccount = registeredCounterAccount;
        }

        [Test]
        public void TestCounterAccount()
        {
            myAccount.MakeTransfer("497164833", new Money());
        }

        [Test]
        [ExpectedException(typeof(BusinessException))]
        public void TestNoCounterAccount()
        {
            CheckingAccount unRegisteredCounterAccount = Accounts.MakeAccount<CheckingAccount>("1439");
            myAccount.MakeTransfer(Accounts.GetAccountNumber(unRegisteredCounterAccount), new Money());
        }

    }
}

File: src/test/java/eu/sig/training/ch04/v3/SavingsAccountTest.java
package eu.sig.training.ch04.v3;

import org.junit.Before;
import org.junit.Test;

import eu.sig.training.ch04.BusinessException;
import eu.sig.training.ch04.Money;

public class SavingsAccountTest {

    SavingsAccount myAccount;
    CheckingAccount registeredCounterAccount;

    @Before
    public void setUp() {
        myAccount = Accounts.makeAccount(SavingsAccount.class, "123456789");
        registeredCounterAccount =
            Accounts.makeAccount(CheckingAccount.class, "497164833");
        myAccount.registeredCounterAccount = registeredCounterAccount;
    }

    @Test
    public void testCounterAccount() throws BusinessException {
        myAccount.makeTransfer("497164833", new Money());
    }

    @Test(expected = BusinessException.class)
    public void testNoCounterAccount() throws BusinessException {
        CheckingAccount unRegisteredCounterAccount =
            Accounts.makeAccount(CheckingAccount.class, "1439");
        myAccount.makeTransfer(
            Accounts.getAccountNumber(unRegisteredCounterAccount), new Money());
    }
}
